File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>>mp;
        vector<vector<string>>ans;
        for(auto i:strs)
        {
            string wrd = i;
            sort(wrd.begin(),wrd.end());
            mp[wrd].push_back(i);
        }
        for(auto j:mp)
        {
            ans.push_back(j.second);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
from typing import List
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ans = defaultdict(list)
        for s in strs:
            cnt = [0]*26
            for c in s:
                cnt[ord(c) - ord("a")] += 1
            ans[tuple(cnt)].append(s)
        return list(ans.values())
        